hazelcast/test/src/ClientTest.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ClientTest.h"
+
+#include <chrono>
+#include <thread>
+
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+
+#include <hazelcast/client/client_config.h>
+#include <hazelcast/client/hazelcast_client.h>
+#include <hazelcast/client/member.h>
+
+#include "HazelcastServerFactory.h"
+#include "remote_controller_client.h"
+
+namespace hazelcast {
+namespace client {
+namespace test {
+
+ClientTest::ClientTest()
+{
+    logger_ = std::make_shared<logger>(
+      "Test", get_test_name(), logger::level::info, logger::default_handler);
+
+    remote_controller_client().ping();
+}
+
+logger&
+ClientTest::get_logger()
+{
+    return *logger_;
+}
+
+std::string
+ClientTest::get_test_name()
+{
+    const auto* info = testing::UnitTest::GetInstance()->current_test_info();
+    std::ostringstream out;
+    out << info->test_case_name() << "_" << info->name();
+    return out.str();
+}
+
+std::string
+ClientTest::get_ca_file_path()
+{
+    return "hazelcast/test/resources/cpp_client.crt";
+}
+
+client_config
+ClientTest::get_config(bool ssl_enabled, bool smart)
+{
+    client_config config;
+
+#ifdef HZ_BUILD_WITH_SSL
+    if (ssl_enabled) {
+        config.set_cluster_name(get_ssl_cluster_name());
+        boost::asio::ssl::context ctx(boost::asio::ssl::context::method::tlsv12_client);
+        ctx.set_default_verify_paths();
+        ctx.load_verify_file(get_ca_file_path());
+
+        config.get_network_config().get_ssl_config().set_context(std::move(ctx));
+    }
+#endif
+
+    config.get_connection_strategy_config().get_retry_config().set_cluster_connect_timeout(
```

Comment:
Why do we set default timeout here ?

Reply:
https://github.com/hazelcast/hazelcast-cpp-client/issues/874#issuecomment-839568694

Reply:
Ok. I leave the decision to you guys. 
Additional note, 5 seconds seems too low to me. I would set it something like a minute to avoid false positives.

Reply:
>  Additional note, 5 seconds seems too low to me. I would set it something like a minute to avoid false positives.

Made it 20. If nightly tests starts failing I'll increase it even more.

Reply:
We have 120secs for assert eventually checks, may be we can have the same timeout for here as well. Or you can wait until it fails. It will be more visible when the compatibility tests run which usually cause timing problems and we may not apply the fix for backwards, hence I would take the safe path and put a higher boundary.

Reply:
it is 120 now. c74ffb76cd4e109fd879ab7718fa6a84b2e6eb1a